tendermint/src/types/time/timestamp.rs

```diff
@@ -238,10 +238,10 @@ impl Timestamp {
     }
 
     /// Creates a new `Timestamp` from the given seconds and nanoseconds.
-    pub fn new(seconds: TimestampSeconds, nanos: Nanoseconds) -> Self {
+    pub const fn new(seconds: TimestampSeconds, nanos: Nanoseconds) -> Self {
         Self {
-            seconds: seconds.into(),
-            nanos: nanos.into(),
+            seconds: seconds.0,
+            nanos: nanos.0 as i32,
```

Comment:
`Nanoseconds` is a structure that contains u32 in range 0-999_999_999, It can't overflow. There are [ways](https://github.com/rumos-io/gears/blob/main/tendermint/src/types/time/timestamp.rs#L169-L192) to construct the value.

Reply:
You can see above that we can cast the value to i32 by `Into::into`. I've updated the code to have a const method for the struct generation.